### ISSUE TITLE
[FEATURE]: Why supervised instruct fine-tuning does not only calculate loss on target tokens?

### ISSUE THREAD

**Issue by sharejing** [hpcaitech/ColossalAI#3330]
### Describe the feature

First of all, thank you for your great work！I'm curious about why not just calculating loss on target tokens. I read BLOOMZ's paper:

![image](https://user-images.githubusercontent.com/33115703/228712084-ba9d1638-35d0-4958-a2de-ff94cb5715ea.png)

From this code snippet, does your supervised instruct fine-tuning seem to be a pre-training method?
![image](https://user-images.githubusercontent.com/33115703/228712292-46300a25-2a85-4842-bb37-0cc6bfc4d3a6.png)
